test/object-store/util/test_file.cpp

```diff
@@ -140,11 +140,14 @@ SyncTestFile::SyncTestFile(std::shared_ptr<realm::SyncUser> user, realm::Schema
     REALM_ASSERT(user);
     sync_config = std::make_shared<realm::SyncConfig>(user, SyncConfig::FLXSyncEnabled{});
     sync_config->stop_policy = SyncSessionStopPolicy::Immediately;
-    sync_config->error_handler = [](std::shared_ptr<SyncSession>, SyncError error) {
-        std::cerr << util::format("An unexpected sync error was caught by the default SyncTestFile handler: '%1'",
-                                  error.message)
+    sync_config->error_handler = [](std::shared_ptr<SyncSession> session, SyncError error) {
+        std::cerr << util::format(
+                         "An unexpected sync error was caught by the default SyncTestFile handler: '%1' for '%2'",
+                         error.message, session->path())
                   << std::endl;
-        abort();
+        if (error.is_fatal) {
+            abort();
```

Comment:
do we want to change this behavior for all our tests? up till now, I think tests that need to handle the is_fatal flag set their own error handlers here so they can capture the specific error conditions they're trying to test. I think it's probably a good thing that we aren't permissive so that our tests can be really explicit about what they're testing.